Dockerfile

```diff
@@ -176,8 +176,22 @@ COPY vendor /go/src/github.com/docker/docker/vendor
 RUN go install -v github.com/cpuguy83/go-md2man \
 	github.com/BurntSushi/toml/cmd/tomlv
 
+# Build the tool for embedding resources in Windows binaries
+ENV RSRC_COMMIT e48dbf1b7fc464a9e85fcec450dddf80816b76e0
+RUN set -x \
+    && git clone https://github.com/akavel/rsrc.git /go/src/github.com/akavel/rsrc \
+    && (cd /go/src/github.com/akavel/rsrc \
+          && git checkout -q $RSRC_COMMIT \
+          && go build -v -o /go/bin/rsrc)
+
 # Wrap all commands in the "docker-in-docker" script to allow nested containers
 ENTRYPOINT ["hack/dind"]
 
 # Upload docker source
 COPY . /go/src/github.com/docker/docker
+
+# Compile the Windows resources into the sources
+RUN rsrc \
+       -manifest /go/src/github.com/docker/docker/hack/make/resources/docker.exe.manifest \
+       -ico /go/src/github.com/docker/docker/hack/make/resources/docker.ico \
+       -o   /go/src/github.com/docker/docker/pkg/winresources/rsrc.syso
```

Comment:
Ah, this part makes me nervous.  This is autogenerated code, which definitely doesn't belong under `pkg`, and should be supported outside the `Dockerfile`-based build, too (ie, when we build natively on a real Windows box).